lms/static/scripts/frontend_apps/components/ErrorDisplay.js

```diff
@@ -84,16 +86,20 @@ export default function ErrorDisplay({ message, error }) {
   return (
     // nb. Wrapper `<div>` here exists to apply block layout to contents.
     <div className="ErrorDisplay">
-      {message && (
-        <p>
-          {!error.message && toSentence(message)}
-          {error.message && (
-            <Fragment>
-              {message}: <i>{toSentence(error.message)}</i>
-            </Fragment>
-          )}
+      {message && error.message && (
+        // Display both error messages if they are both provided
+        <p data-testid="message">
+          {message}: <i>{toSentence(error.message)}</i>
         </p>
       )}
+      {message && !error.message && (
+        // Only the component's message prop is provided
+        <p data-testid="message">{toSentence(message)}</p>
+      )}
+      {!message && error.message && (
+        // Only the error's message property is provided
+        <p data-testid="message">{toSentence(error.message)}</p>
+      )}
```

Comment:
I wonder if we ought to include a case for `!message && !error.message` here for the sake of completeness. This should almost never be triggered in practice because `Error`-like objects should almost always have a `message` property. This could perhaps use the string "An unknown error occurred."

Reply:
Sounds good.